### PR TITLE
src/rgw/rgw_sync_checkpoint.cc: fix a destructed temporary variable access

### PR DESCRIPTION
**Pull request by mhmdawnallah**
This commit adds a static const empty string at `src/rgw/rgw_sync_checkpoint.cc` that lives in the scope of the `empty` function to avoid access a destructed temporary variable.

Fixes: https://tracker.ceph.com/issues/57516
Signed-off-by: Mohamed Awnallah <mohamedmohey2352@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sync_checkpoint.cc [hunk 1]
```diff
@@ -49,8 +49,9 @@ bool operator<(const std::vector<rgw_bucket_shard_sync_info>& lhs,
 
 bool empty(const BucketIndexShardsManager& markers, int size)
 {
+  static const std::string empty_string;
   for (int i = 0; i < size; ++i) {
-    const auto& m = markers.get(i, "");
+    const auto& m = markers.get(i, empty_string);
```
Comment (adamemerson) [inline src/rgw/rgw_sync_checkpoint.cc:54]: I don't think this is necessary? C++ String Literals have static storage duration and exist through the life of the program.

See notes in https://en.cppreference.com/w/cpp/language/string_literal

**Review by adamemerson — CHANGES_REQUESTED** [review]

File: src/rgw/rgw_sync_checkpoint.cc (hunk 1 above)
Comment (yuvalif) [reply src/rgw/rgw_sync_checkpoint.cc:54]: but this is an std::string, not a string literal

**Comment by yuvalif** [comment]
teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2023-03-13_06:02:20-rgw-wip-yuval-test-distro-default-smithi/
5 failures:
* 2 multisite tests (also in baseline)
* sse-kms test. @cbodley are these expected?
```
FAILED s3tests_boto3/functional/test_s3.py::test_sse_kms_multipart_upload - b...
FAILED s3tests_boto3/functional/test_s3.py::test_sse_kms_multipart_invalid_chunks_1
```
* "Cannot create secret" (known failure)
* tempest test issue [here](http://qa-proxy.ceph.com/teuthology/yuvalif-2023-03-13_06:02:20-rgw-wip-yuval-test-distro-default-smithi/7205277/teuthology.log):
```
venv: failed with pass_env values cannot contain whitespace, use comma to have multiple values in a single line, invalid values found 'OS_STDOUT_CAPTURE OS_STDERR_CAPTURE OS_TEST_TIMEOUT OS_TEST_LOCK_PATH TEMPEST_CONFIG TEMPEST_CONFIG_DIR http_proxy HTTP_PROXY https_proxy HTTPS_PROXY no_proxy NO_PROXY ZUUL_CACHE_DIR REQUIREMENTS_PIP_LOCATION GENERATE_TEMPEST_PLUGIN_LIST'
```
* bucket notification test is still running (soon be dead) due to a know  kafka client issue

**Comment by cbodley** [comment]
> teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2023-03-13_06:02:20-rgw-wip-yuval-test-distro-default-smithi/

public link: http://pulpito.ceph.com/yuvalif-2023-03-13_06:02:20-rgw-wip-yuval-test-distro-default-smithi/

> * sse-kms test. @cbodley are these expected?
>
> FAILED s3tests_boto3/functional/test_s3.py::test_sse_kms_multipart_upload - b...
> FAILED s3tests_boto3/functional/test_s3.py::test_sse_kms_multipart_invalid_chunks_1

https://tracker.ceph.com/issues/58751 tracks those 2 failures, specific to the kmip backend (`2-kms/kmip`)

**Review by yuvalif — APPROVED** [review]

**Pull request merged**